Classes/common/query/CDTQIndexUpdater.m

```diff
@@ -138,71 +117,149 @@ - (BOOL)updateIndex:(NSString *)indexName
 }
 
 - (BOOL)updateIndex:(NSString *)indexName
-         withFields:(NSArray /* NSString */ *)fieldNames
-            changes:(TD_RevisionList *)changes
-       lastSequence:(SequenceNumber *)lastSequence
+          fieldNames:(NSArray /* NSString */ *)fieldNames
+    startingSequence:(SequenceNumber)lastSequence
 {
     __block bool success = YES;
 
+    NSString *lastSeqString = [[NSNumber numberWithLongLong:lastSequence] stringValue];
+    CDTFetchChanges *fetcher =
+        [[CDTFetchChanges alloc] initWithDatastore:_datastore startSequenceValue:lastSeqString];
+
+    __weak CDTQIndexUpdater *weakSelf = self;
+
+    NSMutableArray *updateBatch = [NSMutableArray array];
+    NSMutableArray *deleteBatch = [NSMutableArray array];
+
+    fetcher.documentChangedBlock = ^(CDTDocumentRevision *revision) {
+
+        CDTLogVerbose(CDTQ_LOGGING_CONTEXT, @"documentChangedBlock: <%@,%@>", indexName,
+                      revision.docId);
+
+        [updateBatch addObject:revision];
+
+        if (updateBatch.count > 500) {
+            CDTQIndexUpdater *self = weakSelf;
+            if (self) {
```

Comment:
Sorry, my comment was meant to imply using `self`, but it was ambiguous. I changed it.